src/library/scala/collection/immutable/Map.scala

```diff
@@ -204,6 +217,22 @@ object Map extends ImmutableMapFactory[Map] {
       else if (key == key2) new Map2(key1, value1, key2, value)
       else new Map3(key1, value1, key2, value2, key, value)
     def + [V1 >: V](kv: (K, V1)): Map[K, V1] = updated(kv._1, kv._2)
+    override def ++[V1 >: V](xs: GenTraversableOnce[(K, V1)]): Map[K, V1] = ++[(K, V1), Map[K, V1]](xs)(Map.canBuildFrom[K, V1])
+    override def ++[B >: (K, V), That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Map[K, V], B, That]): That = {
+      if (isMapCBF(bf)) that match {
+        case m if m.asInstanceOf[AnyRef] eq EmptyMap => this.asInstanceOf[That]
+        case m: Map1[K, V] => (if(m.contains(key1)) m else m.addTo(this)).asInstanceOf[That]
```

Comment:
Is this right? we're in `Map2`, why does it check only for `key1`?

Also this seems to violate a consistency rule
https://github.com/scala/scala/blob/aeee8f0b2f9f970d07dd33fdb49e93fa1140e103/test/junit/scala/collection/SetMapRulesTest.scala#L30

Reply:
the key1 check is completely wrong, and should be in line with the other MapN classes
I used to have some code like that, but it all got too complex. Maybe it was a bad rebase I did 😲 removed now 

I think it is in line with the consistency rule

m.addTo(this)  calls Map1#addTo, which updates this Map2 by adding calling updated

so - it now reads
```
case m: Map1[K, V] => m.addTo(this).asInstanceOf[That]
case m: Map2[K, V] => m.addTo(this).asInstanceOf[That]
```
which is effectively 
```
case m: Map1[K, V] => (this.updated(m.key1, m.value1)).asInstanceOf[That]
case m: Map2[K, V] => (this.updated(m.key1, m.value1).updated(m.key2, m.value2)).asInstanceOf[That]
```
but with a little less code